application/src/main/java/org/togetherjava/tjbot/commands/moderation/modmail/ModMailCommand.java

```diff
@@ -0,0 +1,177 @@
+        message.queue();
+
+        event.reply("Thank you for contacting the moderators").setEphemeral(true).queue();
+
+    }
+
+    private int getRemainingTime(MessageChannel messageChannel) {
+        return COOLDOWN_DURATION_VALUE - getTimePassedSinceCommandCalled(messageChannel);
+    }
+
+    @NotNull
+    private String getName(@NotNull SlashCommandInteractionEvent event) {
+        String user = event.getUser().getAsMention();
+        boolean optionalMessage = event.getOption(OPTION_MESSAGE_PRIVATE).getAsBoolean();
+        if (optionalMessage) {
+            user = "Anonymous";
+        }
+        return user;
+    }
+
+    private MessageAction buildAttachment(List<ModAuditLogWriter.Attachment> attachments,
+            MessageAction message) {
+        for (ModAuditLogWriter.Attachment attachment : attachments) {
+            message = message.addFile(attachment.getContentRaw(), attachment.name());
+        }
+        return message;
+    }
+
+    @NotNull
+    private MessageEmbed messageEmbed(String user) {
+        return new EmbedBuilder().setAuthor("Modmail Command invoked")
+            .setColor(Color.BLACK)
+            .setTitle("/modmail from '%s' ".formatted(user))
+            .build();
+    }
+
+    @NotNull
+    private List<ModAuditLogWriter.Attachment> getAttachments(
+            @NotNull SlashCommandInteractionEvent event) {
+        String content = event.getOption(MESSAGE).getAsString();
+        List<ModAuditLogWriter.Attachment> attachments = new ArrayList<>();
+        attachments.add(new ModAuditLogWriter.Attachment("content.md", content));
+        return attachments;
+    }
+
+    private @NotNull Optional<TextChannel> getChannel(@NotNull SlashCommandInteractionEvent event) {
+        long guildId = 1004371840245964851L;
+        Guild guild = event.getJDA().getGuildById(guildId);
+        if (guild == null) {
+            return Optional.empty();
+        }
+        return guild.getTextChannelCache()
+            .stream()
+            .filter(channel -> modMailChannelNamePredicate.test(channel.getName()))
+            .findAny();
+    }
+
+    private boolean isChannelOnCooldown(@NotNull MessageChannel channel) {
+        return Optional
+            .ofNullable(channelIdToLastCommandInvocation.getIfPresent(channel.getIdLong()))
+            .map(sinceCommandInvoked -> sinceCommandInvoked.plus(COOLDOWN_DURATION_VALUE,
+                    COOLDOWN_DURATION_UNIT))
+            .filter(Instant.now()::isBefore)
+            .isPresent();
+    }
+
+    private int getTimePassedSinceCommandCalled(@NotNull MessageChannel channel) {
+        return Optional
+            .ofNullable(channelIdToLastCommandInvocation.getIfPresent(channel.getIdLong()))
+            .map(this::getTime)
+            .orElse(COOLDOWN_DURATION_VALUE);
+    }
+
+    private int getTime(Instant sinceCommandInvoked) {
+        return (int) Duration.between(sinceCommandInvoked, Instant.now()).toMinutes();
```

Comment:
why are you casting here?

Reply:
Casting them to int because the return type is long and from the code above

```java
.map(sinceCommandInvoked -> sinceCommandInvoked.plus(COOLDOWN_DURATION_VALUE,
                    COOLDOWN_DURATION_UNIT))
``` 
COOLDOWN_DURATION_UNIT is an int

Reply:
Also wasn't sure, where would it be the best to cast it, so I did it within the method, and since its retrieving minutes, an int return type looked appropriate

Reply:
toMinutes doesn't just render long for no reason

Reply:
Probably for a larger amount of minutes, but ours is 30 minutes max so, it feels like its not a problem but il let you guys decide where to cast it or propose an alternative solution

Reply:
Done